Distro3.1: add imdt connectivity features

tiennguyenuh opened the pull request:
This PR adds imdt connectivity features

LSPgh committed:
imdt-v2h-sbc: Add can-utils

Changes include:
	- Add IMDT V2H-SBC can interface scripts to renesas-core-image-base
	- Align syntax with Styhead release

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

LSPgh committed:
imdt-v2h-sbc: Add Cell Modem

* append difference between this recipe for 8.1.6 and 9.4
* update version of dhcpcd to 9.4.0
* add pico-modem rpm
* align syntax with Styhead release

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

LSPgh committed:
imdt-v2h-sbc: Feature/wifi eth utils

* Ported changes to enable WiFi with IW416 from IMX projects
IW416 requires firmware loading so I have added the firmware
file that we use on our Pico-E. The WiFi driver is built as
a kernel module out-of-tree (it creates mlan.ko, moal.ko).
The recipes in recipes-external have been backported
from the murata upstream layer to work in dunfell.

* Added patch NXP Wlan recipe to build for non IMX platform
* ported from pico-e
* ported from pico-e
* add wireless tools package, update default networks so that none are in adhoc/ap mode
* Align syntax with Styhead release

---------

Co-authored-by: Tendai Makumire <tendai.makumire@imd-tec.com>
Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
imdt-v2h-sbc: add ap1302 firmware

- Add ap1302 firmware for cameras
- Align syntax with Styhead release

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
imdt-v2h-sbc: kernel-module: auto-load moal with modprobe config

Auto-load moal via systemd modules-load.d

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
imdt-v2h-sbc: pin libgpiod to 1.6.5 for imdt-pico-modem compatibility

The imdt-pico-modem firmware requires libgpiod 2.0 API compatibility.
Pin libgpiod to version 1.6.5 to ensure consistent dependency
resolution and avoid rootfs build failures caused by mismatched
libgpiod subpackages.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

vudangRVC commented on conf/templates/rz-cmn/local.conf.sample:
```diff
@@ -377,3 +377,9 @@ BB_PRESSURE_MAX_MEMORY = "100000"
+# imdt-pico-modem depends on the libgpiod 1.x API and is not compatible
+# with libgpiod 2.x. Forcing 1.6.5 avoids rootfs dependency resolution
+# issues caused by mixed libgpiod versions.
+PREFERRED_VERSION_libgpiod = "1.6.5"
```
Do you know if this version is compatible with the other boards?

tiennguyenuh replied:
Yes, other boards are not really affected, it's still using libgpiod but with older version.

vudangRVC commented on include/core-image-renesas-base.inc:
```diff
@@ -18,6 +18,24 @@ IMAGE_INSTALL:append = " \
+    kernel-module-nxp-wlan \
+    murata-binaries \
+    udev-extraconf \
+    wireless-tools \
```
The wireless should be moved to core-sbc. Do you agree?

tiennguyenuh replied:
I agree, all these wireless tools are moved to sbc. Also, I have renamed `core-image-renesas-cmn` to `core-image-renesas-sbc` for consistency

tiennguyenuh committed:
imdt-v2h-sbc: kernel-module-nxp-wlan: add moal lockdep/scan patch fixes

Add two out-of-tree patches for the NXP WLAN driver:
- moal: assign per-instance lockdep class to moal_lock (avoid lockdep false
  recursive locking warning)
- moal: defer cfg80211 scan result reporting from atomic/spinlock context

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

vudangRVC commented on recipes-connectivity/dhcpcd/dhcpcd_9.4.0.bb:
Do you know if OpenEmbedded provides any relevant dhcpcd service?

tiennguyenuh replied:
Actually, this recipe is not needed since we're handling DCHP using systemd-network. So, I have removed this one.

tiennguyenuh committed:
rz-cmn: move wireless package to sbc include file

Changes include:
	- Move wireless package and tools to sbc file
	- Rename `core-image-renesas-cmn.inc` to
	  `core-image-renesas-sbc.inc` for consistency

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
recipes-multimedia: audio: update audio init script

Update audio init for tlv320aic3x codec

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
imdt-v2h-sbc: add fix STA and AP Wi-Fi mode

This commit includes:
- Delete dhcpcd recipe and patches since systemd-networkd handles DHCP
- Improve WiFi utility scripts for better connectivity handling
- Add WiFi virtual interface disabling service to prevent cfg80211 warnings

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>
